packages/x-charts/src/ChartsLegend/DefaultChartsLegend.tsx

```diff
@@ -214,60 +215,49 @@ function DefaultChartsLegend(props: LegendRendererProps) {
       return rep;
     });
 
-    // Move the legend according to padding and position
-    let gapX = 0;
-    let gapY = 0;
+    return [
+      seriesWithRawPosition.map((item) => ({
+        ...item,
+        positionY:
+          item.positionY +
+          (direction === 'row'
+            ? rowMaxHeight[item.rowIndex] / 2 // Get the center of the entire row
+            : item.outerHeight / 2), // Get the center of the item
+      })),
+      totalWidthUsed,
+      totalHeightUsed,
+    ];
```

Comment:
Before it was absolute position. Non I return the size of the legend, and position are relative to the inside of this box.

The `gapX`, `gapY` that are made to align the legend with the border of the chart are moved outside such that we can simply invert

`positionX` to `legendWidth - positionX` to revert line order without having to care about where the legend is in the the chart